sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -366,14 +366,25 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
       fileIndex: FileIndex,
       fs: FileSystem,
       qualifiedPath: Path): Boolean = {
-    val prefixToInvalidate = qualifiedPath.toString
     val needToRefresh = fileIndex.rootPaths
-      .map(_.makeQualified(fs.getUri, fs.getWorkingDirectory).toString)
-      .exists(_.startsWith(prefixToInvalidate))
```

Comment:
I think it'd break Hive's partitioned table. Think about that a partition like `mytable/col=A`, and `col` gets updated?

Reply:
It'd be good to add some comments about this while we're here in any event. It took me a lot of time to review ...

Reply:
I am not sure I understand. Can you give more details on the example above? I the `mytable/col=A` gets updated, why would a rootPath which is not a subdirectory of `mytable/col=A` need to be refreshed?

Reply:
If `col` gets updated, then we need to fresh all subdirectories of `mytable/col=A`. I think this bug fix is correct that subdirectory is not substring of the path string.

Reply:
Actually yeah, I agree.